pupil-spa/src/app/questions-intro/questions-intro.component.ts

```diff
@@ -57,14 +61,7 @@ export class QuestionsIntroComponent implements OnInit, AfterViewInit, OnDestroy
   async onClick() {
     this.auditService.addEntry(new CheckStarted());
     this.clickEvent.emit(null);
-    this.submissionService.submitCheckStartData().toPromise()
-      .then(() => {
-        this.auditService.addEntry(new CheckStartedAPICallSucceeded());
-        this.auditService.addEntry(new CheckStartedApiCalled());
-      })
-      .catch((error) => {
-        this.auditService.addEntry(new CheckStartedApiCalled());
-      });
+    await this.checkStartService.submit();
```

Comment:
I wonder if we actually need to `await` here?

Reply:
@jon-shipley the promise is ignored if we don't await?